### ISSUE TITLE
Difference between `sample` & `rsample`?

### ISSUE THREAD

**Issue by mejliu** [cornellius-gp/gpytorch#764]
Both `model(x)` and `likelihood(model(x))` have `sample()` and `rsample()` functions. However, I have  the following two concerns regarding these.
1. What's the difference between `sample()` and `rsample()`?
2. I know we can draw samples from posterior `likelihood(model(x))`, but how can we draw samples from `model(x)`?  `model(x)` is the prediction for the the function `f`, not output `y`. 

Which one should I use to draw samples from posterior? Thanks.

**Comment by Balandat** [comment]
1. `sample` is literally `rsample` wrapped in `with torch.no_grad()`, so when you don't need gradients.
2. For exact GPs both `model(x)` and `likelihood(model(x))` will be MVNs, so you can sample from either. It depends what you want (i.e. if you want observation noise).
